### PR TITLE
Integrate @hashicorp/react-search into layout

### PR DESCRIPTION
**Pull request by pruett**
This PR updates the UI to expose search capability via our `@hashicorp/react-search` react component and our Algolia-powered index.

Screen Shot 2020-09-01 at 11 58 48 AM

### CODE REVIEW

File: website/components/search-bar/index.jsx [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+        <>
+          <span className="name">
+            <Highlight attribute="page_title" hit={hit} tagName="span" />
+          </span>
+          <span className="description">
+            <Highlight attribute="description" hit={hit} tagName="span" />
+          </span>
+        </>
+      )}
+      placeholder="Search Vault documentation"
```
Comment (pruett) [inline website/components/search-bar/index.jsx:16]:
```suggestion
      placeholder="Search Consul documentation"
```

**Comment by pruett** [comment]
@kylemac @jescalan Now with included `web-components` script utilized at `website/scripts/index_search_content.js`

File: website/components/search-bar/style.css [hunk 2]
```diff
@@ -0,0 +1,8 @@
+.g-search {
+  width: calc(100% - 2rem);
+  max-width: 600px;
+
+  & .c-hits .ais-Highlight-highlighted {
+    /* Add custom background color */
```
Comment (jescalan) [inline website/components/search-bar/style.css:6]: Is this still a TODO?

File: website/package.json [hunk 3]
```diff
@@ -31,21 +32,15 @@
...
-    "dotenv": "8.2.0",
-    "gray-matter": "4.0.2",
     "next": "9.4.4",
     "nuka-carousel": "4.7.0",
     "react": "16.13.1",
     "react-device-detect": "1.13.1",
-    "react-dom": "16.13.1",
-    "remark": "12.0.1",
-    "unist-util-visit": "2.0.3"
+    "react-dom": "16.13.1"
```
Comment (jescalan) [inline website/package.json:40]: The dependency removals are a wonderful thing to see ✨

File: website/scripts/index_search_content.js [hunk 4]
```diff
@@ -1,125 +1,3 @@
...
-        headings.push(title)
-      }
-    })
-  }
-
-  return remark()
-    .use(headingMapper)
-    .process(mdxContent)
-    .then(() => headings)
-}
```
Comment (jescalan) [inline website/scripts/index_search_content.js:125]: 😍

**Review by jescalan — APPROVED** [review]
Ok this looks great to me - just want to make sure that one css file with the comment in it doesn't still need to be addressed?

**Comment by jescalan** [comment]
Also curious why we have env vars on vault but not here? https://github.com/hashicorp/vault/pull/9868/files#diff-ec166e0082218772ecca45a3c2e36871R2

File: website/components/search-bar/style.css (hunk 2 above)
Comment (pruett) [reply website/components/search-bar/style.css:6]: I can prob remove, it was a placeholder for design critiques, but I'll ❌

**Comment by pruett** [comment]
> Also curious why we have env vars on vault but not here? https://github.com/hashicorp/vault/pull/9868/files#diff-ec166e0082218772ecca45a3c2e36871R2

Not sure what you mean. Specifically why do we have an `.env` file?

**Comment by jescalan** [comment]
@pruett afaik the implementation of search across the two sites should be identical, but one has env variables related to search and the other doesn't, right?

**Comment by pruett** [comment]
> @pruett afaik the implementation of search across the two sites should be identical, but one has env variables related to search and the other doesn't, right?

Are you asking about consul's `.env` file (https://github.com/hashicorp/consul/blob/master/website/.env) versus vault's (https://github.com/hashicorp/vault/blob/master/website/.env) ?

**Comment by jescalan** [comment]
Yes, that's what I'm asking about!

**Comment by pruett** [comment]
Both have the following vars:

```
NEXT_PUBLIC_ALGOLIA_APP_ID
NEXT_PUBLIC_ALGOLIA_INDEX
NEXT_PUBLIC_ALGOLIA_SEARCH_ONLY_API_KEY
```

These are expected required both server and client side

**Comment by jescalan** [comment]
I see that - my question was why the env variables changed in one search implementation PR but not the other

**Comment by pruett** [comment]
> I see that - my question was why the env variables changed in one search implementation PR but not the other

Ahhh. Sorry. Yeah, that diff you referenced was indeed a mistake from an earlier effort to add indices to Algolia (even before the UI was using them). All other `.env` files contained the correct index name.

**Pull request merged**